src/scene_map.cpp

```diff
@@ -480,12 +479,29 @@ void Scene_Map::FinishInn() {
 	}
 
 	activate_inn = false;
+	inn_started = false;
 	AsyncNext(std::move(map_async_continuation));
 }
 
 void Scene_Map::UpdateInn() {
-	if (!Audio().BGM_IsPlaying() || Audio().BGM_PlayedOnce()) {
-		Game_System::BgmStop();
-		FinishInn();
+	// Allow message box to render during inn sequence.
+	if (Game_Message::IsMessageVisible()) {
+		Game_Message::Update();
+		return;
 	}
+
+	if (!inn_started) {
+		Transition::instance().InitErase(Transition::TransitionFadeOut, Scene::instance.get());
+		inn_started = true;
+
+		AsyncNext([=]() { StartInn(); });
+		return;
+	}
+
+	if (Audio().BGM_IsPlaying() && Audio().BGM_PlayedOnce()) {
```

Comment:
Not tested but this is not the inverse of the old condition. They inverse would be ``Audio().BGM_IsPlaying() && !Audio().BGM_PlayedOnce()``

Reply:
Fixed, thanks for catching that.